### DISCUSSION TITLE
The difference between useAsyncData and useLazyAsyncData

### DISCUSSION THREAD

**Discussion by Gringo675** [nuxt/nuxt#16596] [Questions]
Hi!
As I can see in all my experiments, there is no difference whether I use useLazyAsyncData or useAsyncData without await. I.e.
```
const { data, pending } = useAsyncData(url, () => $fetch(url))
```
equal
```
const { data, pending } = useLazyAsyncData(url, () => $fetch(url))
```
In both cases we don't wait resolving of async function (so navigation doesn't block), and have pending ref to use in code logic.
In documentation of useAsyncData we have:
`Under the hood, lazy: false uses <Suspense> to block the loading of the route before the data has been fetched.`
But it doesn’t block, if we don’t use await.
And, by the way, I can’t also see any difference between `useLazyAsyncData` or `await useLazyAsyncData`.
Is this normal behavior or I'm missing something? Cause it's looks strange to me.

**Comment by praveshishere** [comment]
Same for me,

**When we use useAsyncData**

1. **At initial load without await:** we wait on server, once resolved then disaplayed on client side with no pending.
2. **At initial load with await:** same as without await
3. **At navigation with await:** wait for data then navigates
4. **At navigation without await:** navigates first with pending, data resolves later

this is just fine, so if need SSR, i use useAsyncData await or not it works same,
and if I want to block client side navigations, I add await

**Now comes useLazyAsyncData:**

At initial load same as **useAsyncData**

only difference is, with or without await, it does not block navigation, data is always resolved after navigation
so if that is the case why do we even need useLazyAsyncData

https://stackblitz.com/edit/nuxt-starter-3mbiq1?file=pages%2Findex.vue

And, also I don't know, how does it work, how can I not add await to a returned promise and get the same type ^^**^^
